File: src/main/java/cn/xunyard/idea/coding/i18n/I18nInspectionSettings.java
package cn.xunyard.idea.coding.i18n;

import cn.xunyard.idea.coding.i18n.logic.I18nInspectionConfiguration;
import com.intellij.openapi.components.*;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2020-01-11
 */
@State(name = "I18nInspectionSettings", storages = {@Storage(StoragePathMacros.WORKSPACE_FILE)})
public class I18nInspectionSettings implements PersistentStateComponent<I18nInspectionConfiguration> {

    private I18nInspectionConfiguration i18nInspectionConfiguration;

    @NotNull
    public static I18nInspectionSettings getInstance(@NotNull Project project) {
        return ServiceManager.getService(project, I18nInspectionSettings.class);
    }

    @Nullable
    @Override
    public I18nInspectionConfiguration getState() {
        return this.i18nInspectionConfiguration;
    }

    @Override
    public void loadState(@NotNull I18nInspectionConfiguration state) {
        this.i18nInspectionConfiguration = state;
    }
}

File: src/main/kotlin/cn/xunyard/idea/coding/i18n/I18nInspectionSettings.kt
package cn.xunyard.idea.coding.i18n

import cn.xunyard.idea.coding.i18n.logic.InspectionConfiguration
import com.intellij.openapi.components.*
import com.intellij.openapi.project.Project

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2020-01-11
 */
@State(name = "I18nInspectionSettings", storages = [Storage(StoragePathMacros.WORKSPACE_FILE)])
class I18nInspectionSettings : PersistentStateComponent<InspectionConfiguration> {
    private var i18nInspectionConfiguration: InspectionConfiguration? = null

    override fun getState(): InspectionConfiguration? {
        return i18nInspectionConfiguration
    }

    override fun loadState(state: InspectionConfiguration) {
        i18nInspectionConfiguration = state
    }

    companion object {
        fun getInstance(project: Project): I18nInspectionSettings {
            return ServiceManager.getService(project, I18nInspectionSettings::class.java)
        }
    }
}